src/commands/image/reingest.ts

```diff
@@ -0,0 +1,132 @@
+import * as colors from 'colors/safe'
+import * as diff from 'diff'
+
+import ApiCommand from '../../base-commands/api'
+import ServiceDiscovery from '../../lib/service-discovery'
+
+export default class ImageReingest extends ApiCommand {
+  static description = 'Reingest an image already present in the images bucket'
+
+  static flags = {
+    ...ApiCommand.flags,
+    help: flags.help({char: 'h'}),
+    dryRun: flags.boolean({
+      name: 'dry-run',
+      description: 'Display the result of the reingestion only',
+      char: 'd'
+    }),
+    compare: flags.boolean({
+      name: 'compare',
+      description: 'Do a dry run, and display the difference between the result and the data returned for this image from media-api',
+      char: 'c'
+    }),
+    force: flags.boolean({
+      name: 'force',
+      description: 'Force reingestion if the image is already present in the Grid',
+      char: 'f'
+    })
+  }
+
+  static args = [{
+    name: 'id',
+    description: 'ID of image',
+    required: true,
+  }]
+
+  async run() {
+    const {args, flags} = this.parse(ImageReingest)
+
+    const imageId: string = args.id
+
+    const http = this.http!
+    const profile = this.profile!
+    const dryRun = flags.dryRun
+    const compare = flags.compare
+    const force = !!flags.force
+
+    const serviceDiscovery = await new ServiceDiscovery(http, profile.mediaApiHost).discover()
+    const adminTools = serviceDiscovery.getLink('admin-tools')
+
+    if (!adminTools) {
+      this.error(`Could not find the admin-tools service. Is it listed at ${profile.mediaApiHost}?`, {exit: 1})
+    }
+
+    const imageResponse = await this.fetchImage(imageId)
+
+    if (this.imageExists(imageResponse) && !force) {
+      this.error(`Cannot reingest - there is already an image with id ${imageId} in the Grid. Add --force to overwrite it.`, {exit: 1})
+    }
+
+    const imageToReingest = await this.fetchProjection(imageId, adminTools.href, !dryRun)
+
+    if (dryRun || compare) {
+      await this.printProjection(imageToReingest, !!compare)
+    } else {
+      this.log(`Reingestion for image with id ${imageId} sent to queue.\n`)
+      try {
+        await this.pollForImage(imageId)
+        this.log('\nImage successfully reingested.')
+      } catch (e) {
+        this.error(e.message)
+      }
+    }
+  }
+
+  private async printProjection(
+    projection: any,
+    diffAgainstES: boolean
+  ) {
+    if (diffAgainstES) {
+      const image = await this.fetchImage(projection.id, '_elasticsearch')
+      const imageDiff = diff.diffJson(image.data || {}, projection)
+      imageDiff
+        .map(this.changeToConsoleString)
+        .map(_ => this.log(_))
+    } else {
+      this.log(JSON.stringify(projection, null, 2))
+    }
+  }
+
+
+  private async fetchProjection(id: string, adminToolsEndpoint: URL, reingest = true) {
+    let endpoint = `${adminToolsEndpoint}/${id}`
+    if (reingest) {
+      endpoint += '?reingest=true'
```

Comment:
minor: would be nice to use a `URLSearchParams` here.

Reply:
Yes! Good spot!